extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -119,39 +141,42 @@ export class SchemaCompareResult {
 				value: localize('schemaCompare.switchLabel', '➔')
 			}).component();
 
+			this.sourceName = this.sourceEndpointInfo ? `${this.sourceEndpointInfo.serverName}.${this.sourceEndpointInfo.databaseName}` : ' ';
 			this.sourceNameComponent = view.modelBuilder.table().withProperties({
 				columns: [
 					{
-						value: sourceName,
+						value: this.sourceName,
 						headerCssClass: 'no-borders',
-						toolTip: sourceName
+						toolTip: this.sourceName
 					},
 				]
 			}).component();
 
 			this.targetNameComponent = view.modelBuilder.table().withProperties({
 				columns: [
 					{
-						value: targetName,
+						value: ' ',
 						headerCssClass: 'no-borders',
-						toolTip: targetName
+						toolTip: ''
```

Comment:
Is this because target is not selected yet? what happens if we open from from pallet i.e. this.sourceName is undefined?

Reply:
yes it will get set after source and target are selected in the dialog. If opened from the command palette, the sourceName gets set to ' ' (there needs to be a space otherwise it shows up as [object object])